tasks/certbot_nginx.yml

```diff
@@ -0,0 +1,189 @@
+  shell: "netstat -antlp | grep LISTEN | grep {{ restund_tls_listen_port }} | grep -v nginx || true"
+  register: tls_port_use
+  changed_when: false
+
+- fail:
+    msg: "restund is listening on port {{ restund_tls_listen_port }}, which should be used by nginx since certbot_enabled = true. Please stop the restund process before continuing, or set certbot_enabled to false."
+  when: tls_port_use.stdout != ""
+
+# 1.
+
+- name: Install software-properties-common
+  apt:
+    cache_valid_time: 86400
+    pkg:
+    - software-properties-common
+
+- name: add universe repository
+  apt_repository:
+    repo: "{{ item }}"
+    state: present
+  loop:
+    - "deb http://archive.ubuntu.com/ubuntu/ {{ ansible_distribution_release }} universe"
+    - "deb http://archive.ubuntu.com/ubuntu/ {{ ansible_distribution_release }} universe"
+    - "deb http://security.ubuntu.com/ubuntu/ {{ ansible_distribution_release }} universe"
+
+- name: add repository
+  apt_repository:
+    repo: ppa:certbot/certbot
+    state: present
+
+- name: Install certbot and dig
+  apt:
+    update_cache: true
+    pkg:
+    - dnsutils # for 'dig'
+    - certbot
+
+# 2.
+
+- name: gather DNS info using dig
+  shell: "dig +short {{ certbot_domain }}"
+  register: dig_result
+  when: certbot_enable_checks
+  changed_when: false
+
+- name: Fail if certbot generation cannot succeed due to missing DNS
+  fail:
+    msg: "In order to use the HTTP01 certbot challenge, {{ certbot_domain }} needs to point to (one of) this machine's IP. Please configure your DNS. Note: Set certbot_enable_checks: false to disable this check in case you are using load balancers fronting this machine. Result of DNS query: {{ dig_result }}"
+  when: certbot_enable_checks and dig_result.stdout not in ansible_all_ipv4_addresses
+
+# 3.
+
+- name: create nginx user
+  user:
+    name: nginx
+
+- name: install / configure nginx using role (but without custom configurations)
+  include_role:
+    name: nginx
+  vars:
+    nginx_debug_output: true
+    nginx_branch: stable
+
+- name: check nginx open port 80 (needed for cert generation & renewal)
+  wait_for:
+    port: 80
+    timeout: 5
+
+# 4.
+
+- name: check cert exists
+  stat:
+    path: "/etc/letsencrypt/live/{{ certbot_domain }}/fullchain.pem"
+  register: cert_file
+
+- name: create certificates with cerbot/letsencrypt
+  shell: 'certbot certonly --webroot -w /var/www/html -n --agree-tos --email {{ certbot_admin_email }} -d {{ certbot_domain }}'
+  when: not cert_file.stat.exists
+
+
+# 5.
+
+- name: install / configure nginx using role
+  include_role:
+    name: nginx
+  vars:
+    nginx_debug_output: true
+    nginx_branch: stable
+    nginx_state: latest
+
+    nginx_logrotate_conf_enable: true
+
+    nginx_main_template_enable: true
+    nginx_main_template:
+      worker_connections: 1024 # higher? TODO
+      stream_enable: true
+      user: nginx
+      pid: /var/run/nginx.pid
+      worker_processes: auto
+      worker_rlimit_nofile: 65535
```

Comment:
Not sure if I understand this correctly, but would it make sense to cut this in half (see https://stackoverflow.com/questions/37591784/nginx-worker-rlimit-nofile)? Or what was your thinking behind this number other than max open ports?

Reply:
Good catch; actually the default nginx role doesn't set an adequate number of allowed file descriptors. Changed things in f7412df9f1fb77c51d5bde8645cc55f203d2d562